internal/api/download.go

```diff
@@ -0,0 +1,91 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"net/http"
+)
+
+func (c *ApiClient) Download(ctx Context, reportType string, reportJournalType string, reportScheduleType string, reportAccountType string, reportDebtType string, dateOfTransaction string, dateFrom string, dateTo string, email string) error {
+	var body bytes.Buffer
+	var dateTransformed *model.Date
+	var toDateTransformed *model.Date
+	var fromDateTransformed *model.Date
+
+	if dateOfTransaction != "" {
+		raisedDateFormatted := model.NewDate(dateOfTransaction)
+		dateTransformed = &raisedDateFormatted
+	}
+
+	if dateTo != "" {
+		startDateFormatted := model.NewDate(dateTo)
+		toDateTransformed = &startDateFormatted
+	}
+
+	if dateFrom != "" {
+		endDateFormatted := model.NewDate(dateFrom)
+		fromDateTransformed = &endDateFormatted
+	}
```

Comment:
I think you could extract this logic to a `NewDownload` function, which would simplify the `(*ApiClient).Download` method signature a lot

Reply:
done